modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/StereotypesDeclarationEnricher.java

```diff
@@ -0,0 +1,168 @@
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.NestedChainDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.SourceDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.WithOperationsDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.WithSourcesDeclaration;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.api.meta.model.stereotype.StereotypeModel;
+import org.mule.runtime.api.meta.model.stereotype.StereotypeModelBuilder;
+import org.mule.runtime.core.api.util.ClassUtils;
+import org.mule.runtime.extension.api.annotation.error.ErrorTypes;
+import org.mule.runtime.extension.api.annotation.error.Throws;
+import org.mule.runtime.extension.api.annotation.param.stereotype.AllowedStereotypes;
+import org.mule.runtime.extension.api.annotation.param.stereotype.Stereotype;
+import org.mule.runtime.extension.api.declaration.fluent.util.IdempotentDeclarationWalker;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.extension.api.loader.DeclarationEnricher;
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.extension.api.runtime.process.Chain;
+import org.mule.runtime.extension.api.stereotype.MuleStereotypeDefinition;
+import org.mule.runtime.extension.api.stereotype.MuleStereotypeFactory;
+import org.mule.runtime.extension.api.stereotype.StereotypeDefinition;
+import org.mule.runtime.module.extension.internal.loader.java.property.ImplementingMethodModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.property.ImplementingTypeModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.type.ExtensionParameter;
+import org.mule.runtime.module.extension.internal.loader.java.type.runtime.MethodWrapper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * {@link DeclarationEnricher} implementation which enriches the {@link ExtensionModel} and their {@link OperationModel} from the
+ * used {@link ErrorTypes} and {@link Throws} in an Annotation based extension.
+ *
+ * @since 4.0
+ */
+public class StereotypesDeclarationEnricher implements DeclarationEnricher {
+
+  @Override
+  public void enrich(ExtensionLoadingContext extensionLoadingContext) {
+    new EnricherDelegate().apply(extensionLoadingContext);
+  }
+
+  private static class EnricherDelegate {
+
+    private final Map<StereotypeDefinition, StereotypeModel> stereotypes = new HashMap<>();
+
+    public void apply(ExtensionLoadingContext extensionLoadingContext) {
+      ExtensionDeclarer extensionDeclarer = extensionLoadingContext.getExtensionDeclarer();
```

Comment:
this method must be completely executed using the extension's class loader

Reply:
done